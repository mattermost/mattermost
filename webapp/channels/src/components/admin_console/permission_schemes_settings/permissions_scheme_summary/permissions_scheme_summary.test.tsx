// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {shallow} from 'enzyme';
import React from 'react';

import PermissionsSchemeSummary from 'components/admin_console/permission_schemes_settings/permissions_scheme_summary/permissions_scheme_summary';
import ConfirmModal from 'components/confirm_modal';

describe('components/admin_console/permission_schemes_settings/permissions_scheme_summary', () => {
    const defaultProps = {
        scheme: {
            id: 'id',
            name: 'xxxxxxxxxx',
            display_name: 'Test',
            description: 'Test description',
        },
        teams: [
            {id: 'xxx', name: 'team-1', display_name: 'Team 1'},
            {id: 'yyy', name: 'team-2', display_name: 'Team 2'},
            {id: 'zzz', name: 'team-3', display_name: 'Team 3'},
        ],
        actions: {
            deleteScheme: jest.fn().mockResolvedValue({data: true}),
        },
    } as any;

    test('should match snapshot on default data', () => {
        const wrapper = shallow(
            <PermissionsSchemeSummary {...defaultProps}/>,
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot on more than eight teams', () => {
        const wrapper = shallow(
            <PermissionsSchemeSummary
                {...defaultProps}
                teams={[
                    {id: 'aaa', name: 'team-1', display_name: 'Team 1'},
                    {id: 'bbb', name: 'team-2', display_name: 'Team 2'},
                    {id: 'ccc', name: 'team-3', display_name: 'Team 3'},
                    {id: 'ddd', name: 'team-4', display_name: 'Team 4'},
                    {id: 'eee', name: 'team-5', display_name: 'Team 5'},
                    {id: 'fff', name: 'team-6', display_name: 'Team 6'},
                    {id: 'ggg', name: 'team-7', display_name: 'Team 7'},
                    {id: 'hhh', name: 'team-8', display_name: 'Team 8'},
                    {id: 'iii', name: 'team-9', display_name: 'Team 9'},
                    {id: 'jjj', name: 'team-9', display_name: 'Team 10'},
                ]}
            />,
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot on no teams', () => {
        const wrapper = shallow(
            <PermissionsSchemeSummary
                {...defaultProps}
                teams={[]}
            />,
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('should ask to toggle on row toggle', () => {
        const deleteScheme = jest.fn().mockResolvedValue({data: true});
        const wrapper = shallow(
            <PermissionsSchemeSummary
                {...defaultProps}
                actions={{
                    deleteScheme,
                }}
            />,
        );
        expect(deleteScheme).not.toHaveBeenCalled();
        wrapper.find('.delete-button').first().simulate('click', {stopPropagation: jest.fn()});
        expect(deleteScheme).not.toHaveBeenCalled();
        wrapper.find(ConfirmModal).first().props().onCancel?.(true);
        expect(deleteScheme).not.toHaveBeenCalled();

        wrapper.find('.delete-button').first().simulate('click', {stopPropagation: jest.fn()});
        wrapper.find(ConfirmModal).first().props().onConfirm?.(true);
        expect(deleteScheme).toHaveBeenCalledWith('id');
    });
});
