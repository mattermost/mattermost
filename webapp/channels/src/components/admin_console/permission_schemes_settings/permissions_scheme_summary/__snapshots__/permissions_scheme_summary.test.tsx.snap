// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/admin_console/permission_schemes_settings/permissions_scheme_summary should match snapshot on default data 1`] = `
<div
  className="permissions-scheme-summary"
  data-testid="permissions-scheme-summary"
  onClick={[Function]}
>
  <div
    onClick={[Function]}
  >
    <ConfirmModal
      confirmButtonClass="btn btn-primary"
      confirmButtonText={
        <Memo(LoadingWrapper)
          loading={false}
          text={
            Object {
              "defaultMessage": "Deleting...",
              "id": "admin.permissions.permissionsSchemeSummary.deleting",
            }
          }
        >
          <Memo(MemoizedFormattedMessage)
            defaultMessage="Yes, Delete"
            id="admin.permissions.permissionsSchemeSummary.deleteConfirmButton"
          />
        </Memo(LoadingWrapper)>
      }
      message={
        <div>
          <p>
            <Memo(MemoizedFormattedMessage)
              defaultMessage="The permissions in the teams using this scheme will reset to the defaults in the System Scheme. Are you sure you want to delete the {schemeName} scheme?"
              id="admin.permissions.permissionsSchemeSummary.deleteConfirmQuestion"
              values={
                Object {
                  "schemeName": "Test",
                }
              }
            />
          </p>
        </div>
      }
      modalClass=""
      onCancel={[Function]}
      onConfirm={[Function]}
      show={false}
      title={
        <Memo(MemoizedFormattedMessage)
          defaultMessage="Delete {scheme} scheme?"
          id="admin.permissions.permissionsSchemeSummary.deleteSchemeTitle"
          values={
            Object {
              "scheme": "Test",
            }
          }
        />
      }
    />
  </div>
  <div
    className="permissions-scheme-summary--header"
  >
    <div
      className="title"
    >
      Test
    </div>
    <div
      className="actions"
    >
      <Link
        className="edit-button"
        data-testid="Test-edit"
        to="/admin_console/user_management/permissions/team_override_scheme/id"
      >
        <MemoizedFormattedMessage
          defaultMessage="Edit"
          id="admin.permissions.permissionsSchemeSummary.edit"
        />
      </Link>
      -
      <a
        className="delete-button"
        data-testid="Test-delete"
        onClick={[Function]}
      >
        <MemoizedFormattedMessage
          defaultMessage="Delete"
          id="admin.permissions.permissionsSchemeSummary.delete"
        />
      </a>
    </div>
  </div>
  <div
    className="permissions-scheme-summary--description"
  >
    Test description
  </div>
  <div
    className="permissions-scheme-summary--teams"
  >
    <span
      className="team"
      key="xxx"
    >
      Team 1
    </span>
    <span
      className="team"
      key="yyy"
    >
      Team 2
    </span>
    <span
      className="team"
      key="zzz"
    >
      Team 3
    </span>
  </div>
</div>
`;

exports[`components/admin_console/permission_schemes_settings/permissions_scheme_summary should match snapshot on more than eight teams 1`] = `
<div
  className="permissions-scheme-summary"
  data-testid="permissions-scheme-summary"
  onClick={[Function]}
>
  <div
    onClick={[Function]}
  >
    <ConfirmModal
      confirmButtonClass="btn btn-primary"
      confirmButtonText={
        <Memo(LoadingWrapper)
          loading={false}
          text={
            Object {
              "defaultMessage": "Deleting...",
              "id": "admin.permissions.permissionsSchemeSummary.deleting",
            }
          }
        >
          <Memo(MemoizedFormattedMessage)
            defaultMessage="Yes, Delete"
            id="admin.permissions.permissionsSchemeSummary.deleteConfirmButton"
          />
        </Memo(LoadingWrapper)>
      }
      message={
        <div>
          <p>
            <Memo(MemoizedFormattedMessage)
              defaultMessage="The permissions in the teams using this scheme will reset to the defaults in the System Scheme. Are you sure you want to delete the {schemeName} scheme?"
              id="admin.permissions.permissionsSchemeSummary.deleteConfirmQuestion"
              values={
                Object {
                  "schemeName": "Test",
                }
              }
            />
          </p>
        </div>
      }
      modalClass=""
      onCancel={[Function]}
      onConfirm={[Function]}
      show={false}
      title={
        <Memo(MemoizedFormattedMessage)
          defaultMessage="Delete {scheme} scheme?"
          id="admin.permissions.permissionsSchemeSummary.deleteSchemeTitle"
          values={
            Object {
              "scheme": "Test",
            }
          }
        />
      }
    />
  </div>
  <div
    className="permissions-scheme-summary--header"
  >
    <div
      className="title"
    >
      Test
    </div>
    <div
      className="actions"
    >
      <Link
        className="edit-button"
        data-testid="Test-edit"
        to="/admin_console/user_management/permissions/team_override_scheme/id"
      >
        <MemoizedFormattedMessage
          defaultMessage="Edit"
          id="admin.permissions.permissionsSchemeSummary.edit"
        />
      </Link>
      -
      <a
        className="delete-button"
        data-testid="Test-delete"
        onClick={[Function]}
      >
        <MemoizedFormattedMessage
          defaultMessage="Delete"
          id="admin.permissions.permissionsSchemeSummary.delete"
        />
      </a>
    </div>
  </div>
  <div
    className="permissions-scheme-summary--description"
  >
    Test description
  </div>
  <div
    className="permissions-scheme-summary--teams"
  >
    <span
      className="team"
      key="aaa"
    >
      Team 1
    </span>
    <span
      className="team"
      key="bbb"
    >
      Team 2
    </span>
    <span
      className="team"
      key="ccc"
    >
      Team 3
    </span>
    <span
      className="team"
      key="ddd"
    >
      Team 4
    </span>
    <span
      className="team"
      key="eee"
    >
      Team 5
    </span>
    <span
      className="team"
      key="fff"
    >
      Team 6
    </span>
    <span
      className="team"
      key="ggg"
    >
      Team 7
    </span>
    <span
      className="team"
      key="hhh"
    >
      Team 8
    </span>
    <WithTooltip
      title="Team 9, Team 10"
    >
      <span
        className="team"
        key="extra-teams"
      >
        <MemoizedFormattedMessage
          defaultMessage="+{number} more"
          id="admin.permissions.permissionsSchemeSummary.moreTeams"
          values={
            Object {
              "number": 2,
            }
          }
        />
      </span>
    </WithTooltip>
  </div>
</div>
`;

exports[`components/admin_console/permission_schemes_settings/permissions_scheme_summary should match snapshot on no teams 1`] = `
<div
  className="permissions-scheme-summary"
  data-testid="permissions-scheme-summary"
  onClick={[Function]}
>
  <div
    onClick={[Function]}
  >
    <ConfirmModal
      confirmButtonClass="btn btn-primary"
      confirmButtonText={
        <Memo(LoadingWrapper)
          loading={false}
          text={
            Object {
              "defaultMessage": "Deleting...",
              "id": "admin.permissions.permissionsSchemeSummary.deleting",
            }
          }
        >
          <Memo(MemoizedFormattedMessage)
            defaultMessage="Yes, Delete"
            id="admin.permissions.permissionsSchemeSummary.deleteConfirmButton"
          />
        </Memo(LoadingWrapper)>
      }
      message={
        <div>
          <p>
            <Memo(MemoizedFormattedMessage)
              defaultMessage="The permissions in the teams using this scheme will reset to the defaults in the System Scheme. Are you sure you want to delete the {schemeName} scheme?"
              id="admin.permissions.permissionsSchemeSummary.deleteConfirmQuestion"
              values={
                Object {
                  "schemeName": "Test",
                }
              }
            />
          </p>
        </div>
      }
      modalClass=""
      onCancel={[Function]}
      onConfirm={[Function]}
      show={false}
      title={
        <Memo(MemoizedFormattedMessage)
          defaultMessage="Delete {scheme} scheme?"
          id="admin.permissions.permissionsSchemeSummary.deleteSchemeTitle"
          values={
            Object {
              "scheme": "Test",
            }
          }
        />
      }
    />
  </div>
  <div
    className="permissions-scheme-summary--header"
  >
    <div
      className="title"
    >
      Test
    </div>
    <div
      className="actions"
    >
      <Link
        className="edit-button"
        data-testid="Test-edit"
        to="/admin_console/user_management/permissions/team_override_scheme/id"
      >
        <MemoizedFormattedMessage
          defaultMessage="Edit"
          id="admin.permissions.permissionsSchemeSummary.edit"
        />
      </Link>
      -
      <a
        className="delete-button"
        data-testid="Test-delete"
        onClick={[Function]}
      >
        <MemoizedFormattedMessage
          defaultMessage="Delete"
          id="admin.permissions.permissionsSchemeSummary.delete"
        />
      </a>
    </div>
  </div>
  <div
    className="permissions-scheme-summary--description"
  >
    Test description
  </div>
  <div
    className="permissions-scheme-summary--teams"
  />
</div>
`;
